Documentation for CPU Profiling tools (#2666)
o3de/o3de.org: content/docs/user-guide/profiling/cpu_profiling/_index.md

![Imgui profiler search](/images/user-guide/profiling/imgui/profile-event-search.jpg)

When the timeline is scaled-out, you will be able to see the frametime in ms for every captured frame in the profiler. This is an easy way to find frames which are over your budget.

### Save your capture

Press the "Capture" button to save your captured data in a json file. The path will be indicated right above the button. You can load any previous capture thanks to the "Load file" button.

![Imgui profiler save](/images/user-guide/profiling/imgui/save.png)

## Prerequisites for Tracy, Optick and Superluminal

These profiler Gems are located in the [o3de-extras](https://github.com/o3de/o3de-extras) github repository. 

1. This repository follows the Remote Content format, you can follow [this documentation](/docs/user-guide/remote-content/use-a-remote-repository) to learn how to **register a remote repository**.

2. Else if you prefer to download o3de-extra yourself, you can follow [this documentation](/docs/user-guide/project-config/register-gems). Within o3de-extra folder, the profiler gems are located in "*Gems/ExternalProfilers*".

> Once that the profiler gem that you want to use has been registered, you need to enable it for your project and rebuild the project. Refer to the [first steps of enabling the Built-In profiler](#enabling-the-built-in-profiler-gem) to see how to enable a gem for your project.

Reviewer: ```suggestion
Once you have registered a profiler gem, enable it and rebuild the project. Refer to the [first steps of enabling the Built-In profiler](#enabling-the-built-in-profiler-gem) to see how to enable a gem for your project.
```

> ## Using Tracy profiler (All platforms)

Reviewer: From this section down, how much of this is already covered by the docs for the respective profilers? It's a general rule of thumb to not document the general usage of third party applications. What we document here should only be where the profilers intersect with O3DE.

Author: The issue is that the documentation for these tools is either nonexistant or too obtuse for newcommer, I try to give broad strokes on how to launch them and get quick result, but feel free to pinpoint an area if you think it is too detailed

Author: If you take tracy for example, this is the documentation pdf, I don't expect a lot of people do dig into it https://github.com/wolfpld/tracy/releases/latest/download/tracy.pdf

Reviewer: Alright, yeah, that's what I kind of expected. This would be one of the exceptions to the rule of thumb, then. I'll get it reviewed and edited shortly.
